instructor/patch.py

```diff
@@ -92,11 +92,16 @@ def handle_response_model(
         elif mode == Mode.JSON or mode == Mode.MD_JSON:
             if mode == Mode.JSON:
                 new_kwargs["response_format"] = {"type": "json_object"}
-                # check that the first message is a system message
-                # if it is not, add a system message to the beginning
-                message = f"""Make sure that your response to any message matches the json_schema below,
-                        do not deviate at all: \n{response_model.model_json_schema()['properties']}
-                        """
+                # Check for nested models
+                if response_model.model_json_schema()['$defs']:
```

Comment:
can we do something like 

mesges = f" .. response_model.model_json_schema()"

if "defs" in response_model.model_json_schema():
    messages += "\nhere are some more definitions to adhere too:\n{...}"
    
   I wonder how clean it can be if we just did `response_model.model_json_schema()`

Reply:
Done!